### ISSUE TITLE
Icon and name on avatar

### ISSUE THREAD

**Issue by billba** [Microsoft/BotFramework-WebChat#43]
What would we do with them? Is it important to ape the old control?

**Comment by billba** [comment]
Based on conversation it seems we do need bot-name, which might also be the name of a participant.

**Comment by kevkor** [comment]
Yes, we do need bot name and also the bot icon (which is on by default, overrideable, and optional). See mocks in https://github.com/billba/botchat/issues/50

**Comment by danmarshall** [comment]
Should we split this into 2 new issues? I can implement the icon in the CSS.

**Comment by billba** [comment]
This is now fully supported in the props and the state, you should be able to style it.

**Comment by ubreddy** [comment]
bot icon and name still does not come up in the webchannel of botframework.

**Comment by billba** [comment]
Yup, still not implemented in the styling.

**Comment by rohantare** [comment]
Yes, the chat bot icon does not appear in webchat control. Any idea when this would be in?

**Comment by billba** [comment]
@danmarshall you guys getting ready to add icons back in?

**Comment by danmarshall** [comment]
I'm guessing about a week or so at least.

**Comment by ubreddy** [comment]
Any update on this? was expecting this issue to be closed this week based on @danmarshall  comment.

**Comment by danmarshall** [comment]
Hi folks, I'm a little behind on this, but I'm glad to know your interest. This work item has been bumped up on my priority list. Watch the repo for a related pull request. Thanks!

**Comment by danmarshall** [comment]
Concept art in #410

**Comment by ubreddy** [comment]
Any update on bot icon and welcome message appearing on web channel by default as it was available in earlier version?

**Comment by danmarshall** [comment]
Hi @ubreddy - let's separate the two issues, for your"welcome message" issue, I believe this is covered in #173 . Here we can discuss the icon feature. As it turns out, this feature is not quite as simple as it appears.
For example, to enable group conversation, we need a strategy for this icon to be distinct for each user. So we are working out the complexities, and can have a release once we have rationalized them. For an early preview you can take a look at [this fork](https://github.com/Fignewton5/BotFramework-WebChat). 

Remember you can always build your own WebChat with your own customizations :)

**Comment by CabuxaMapache** [comment]
I'm really interested in showing the bot icon, but It's not present in the activity.from prop. In the fork mentioned by @danmarshall the bot icon url is not part of the user props...why?

**Comment by danmarshall** [comment]
Hi @CabuxaMapache we've still got some design to work through this in DirectLine. Stay tuned, I'll be getting to this after we finish up AdaptiveCards. Appreciate your patience!

**Comment by 20112077** [comment]
Hi @danmarshall,
do you have any idea when the bot icon will be probably available in the web chat?

**Comment by vinod83k** [comment]
Hi @danmarshall , we are also looking to display bot icon in web chat. Please suggest as when this feature will be available..

**Comment by assaf-cream** [comment]
Hi,
I want to make sure I'm following the correct discussion - is the requirement here addresses adding an icon next to the bot messages in the conversation flow?

I would also like to know if there is an option to add icon next to the user side of the discussion (could be some general icon in case of an unrecognized user)

**Comment by 20112077** [comment]
@danmarshall is there any information when the bot icon will be available? I have a customer waiting for it?

**Comment by narulasrinivas** [comment]
Hi...What is the planned to release this ?

**Comment by danmarshall** [comment]
Hello everyone. We can't give out a date, but you may want to look at [this fork](https://github.com/Fignewton5/BotFramework-WebChat) to get you started. As I've said above, to do this feature correctly is going to require changes in DirectLine itself.

I will update this thread when we have more news.

**Comment by TestUsetGitHubUser** [comment]
Do you have any update on this  @danmarshall ?

**Comment by billba** [comment]
@TestUsetGitHubUser it's bad form to keep pinging on an open issue. @danmarshall specifically said he'd update this thread when we have more news. You can take him at his word.

On the other hand, if you want to signal your desire for this feature request, you can upvote a relevant message as others have done for [this one](https://github.com/Microsoft/BotFramework-WebChat/issues/43#issuecomment-331417898).

**Comment by filipyoo** [comment]
Is [this fork and branch](https://github.com/tonyanziano/BotFramework-WebChat/tree/v-ananzi/bot-avatar) a usable fork ? If yes, why is it not merged with master and is in 'stale' state ? :o

**Comment by billba** [comment]
@filipyoo Last I checked there are 374 forks out there with I don't know how many branches. Why are you asking about this particular one?

**Comment by filipyoo** [comment]
@billba It was mentioned [here](https://github.com/Microsoft/BotFramework-WebChat/issues/43#issuecomment-331526120) by @danmarshall, and it seems to work but not yet implemented officially.

**Comment by DanielTizon** [comment]
I can't believe... this issue is open for almost 2 years, and still there isn't bot icon. 

It is a so basic feature, that I'm really surprised.

is Microsoft leaving this project or Microsoft haven't enough cash to hire devs?

**Comment by TestUsetGitHubUser** [comment]
@DanielTizon  I added inside botchat script manually it works for me.

**Comment by DanielTizon** [comment]
@TestUsetGitHubUser what script? It is a long 2-years old conversation with a lot of links.

**Comment by TestUsetGitHubUser** [comment]
Download this script locally  <script src="https://cdn.botframework.com/botframework-webchat/latest/botchat.js"></script>
And find this code below and create this img element.
>  if (r == "bot")
>  { return o.createElement("div", { "data-activity-id": this.props.activity.id, className: i, onClick: this.props.onClickActivity },
>  **o.createElement("img", { className: "botIcon" + r, src: "Images/xx.jpg", width: "28px", height: "32px", float: "left" })**, 
> o.createElement("div", { className: "wc-message wc-message-from-" + r, ref: function (t) { return e.messageDiv = t } }, 
> o.createElement("div", { className: s }, 
> o.createElement("svg", { className: "wc-message-callout" },
>  o.createElement("path", { className: "point-left", d: "m0,6 l6 6 v-12 z" }), 
> o.createElement("path", { className: "point-right", d: "m6,6 l-6 6 v-12 z" })), this.props.children)),
```
   
 Download ...botchat.cssand  and add .botIcon class and style it as you want like:
```
.botIcon {
    float: left !important;
    border-radius: 50%;
    height: 39px !important;
    width: 39px !important;
}
```

**Comment by michel-weber** [comment]
@TestUsetGitHubUser I used your hack to implement it. What a shame that this very basic functionality has not been implemented yet.

**Comment by compulim** [comment]
The work is done in PR #1486.

**Comment by compulim** [comment]
The PR is being merged and it will be resolved by `@4.2.1-master.d35a7ac` and `>= @4.2.1`.
